sda-commons-dependency-check/build.gradle

```diff
@@ -9,4 +12,19 @@ dependencies {
   api project(":sda-commons-asyncapi")
 
   testImplementation "io.github.classgraph:classgraph"
+}
+
+/**
+ * A rule that forces all bouncycastle modules to use the same version.
+ */
```

Comment:
tbh, I have no idea what this code is doing, so I have to rely on the documentation.

Reply:
I copied it from https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-dependency-check/build.gradle

It's a pretty straightforward copy of https://docs.gradle.org/current/userguide/dependency_version_alignment.html#sec:align_bom_platform

TBH I'm not sure if it works perfectly because it will create a virtual platform for every version that is found for bouncycastle. But I double-checked in IntelliJ that all artifacts from BC now use 1.76 as version.

Reply:
I wonder how code in the check module can _force_ dependencies for the consumers. Consumers won't have a dependency to this module. That's why I'm asking if it's rather a _verify_. I would expect a _force_ in the dependencies module. Did the dependency tree (in the lib and maybe also in consuming services) change with this rule?

Reply:
Wenn I change the version of one single BC dependency in sda-commons-dependencies (in order to have different versions in the project), I get the following error for `./gradlew :sda-commons-dependency-check:build`

```
Execution failed for task ':sda-commons-dependency-check:compileTestJava'.
> Could not resolve all files for configuration ':sda-commons-dependency-check:testCompileClasspath'.
   > Could not resolve org.bouncycastle:bcpkix-jdk18on:1.76.
     Required by:
         project :sda-commons-dependency-check > project :sda-commons-dependencies
      > Multiple forces on different versions for virtual platform org.bouncycastle:bouncycastle-virtual-platform
   > Could not resolve org.bouncycastle:bcprov-jdk18on:1.74.
     Required by:
         project :sda-commons-dependency-check > project :sda-commons-dependencies
      > Multiple forces on different versions for virtual platform org.bouncycastle:bouncycastle-virtual-platform
```